lib/charms/ip_router_interface/v0/ip_router_interface.py

```diff
@@ -1,23 +1,92 @@
-"""TODO: Add a proper docstring here.
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
 
-This is a placeholder docstring for this charm library. Docstrings are
-presented on Charmhub and updated whenever you push a new version of the
-library.
+"""Library for the ip-router integration
 
-Complete documentation about creating and documenting libraries can be found
-in the SDK docs at https://juju.is/docs/sdk/libraries.
+This library contains the Requires and Provides classes for interactions through the 
+ip-router interface
 
-See `charmcraft publish-lib` and `charmcraft fetch-lib` for details of how to
-share and consume charm libraries. They serve to enhance collaboration
-between charmers. Use a charmer's libraries for classes that handle
-integration with their charm.
+## Getting Started
 
-Bear in mind that new revisions of the different major API versions (v0, v1,
-v2 etc) are maintained independently.  You can continue to update v0 and v1
-after you have pushed v3.
+From a charm directory, fetch the library using `charmcraft`:
 
-Markdown is supported, following the CommonMark specification.
-"""
+```shell
+charmcraft fetch-lib charms.ip_router_interface.v0.ip_router_interface TODO: Verify this works
+```
+
+### Provider charm
+This example provider charm is all we need to listen to ip-router requirer requests:
```

Comment:
I think this interface is a bit confusing, and would like to see clearer documentation about what we expect the Provider and the Requirer to do. By that, I mean to clarify that the Provider is expected to be the router, and should implement the routes it receives. Same thing for the Requirer. In the examples, I would like to see some concrete usage of the routes. As it is, the example provider does nothing with the routes.

Reply:
I expanded the documentation a bit, please take a look again

Reply:
This is better, thank you!